tests/pkg-layering/main.yml

```diff
@@ -0,0 +1,319 @@
+---
+# vim: set ft=ansible:
+#
+# !!!NOTE!!! This playbook was tested using Ansible 2.1; it is recommended
+# that the same version is used.
+#
+# This playbook is actually multiple playbooks contained in a single file.
+# I had to take this approach, because I encountered difficulties re-using
+# the same roles with non-unique variables.  Sometimes the roles would be
+# executed with the use of the 'always' tag, sometimes not.
+#
+# The wrinkle in this approach is that sharing global variables across the
+# multiple playbooks requires the use of a file containing the variables
+# which is included in each playbook with the 'vars_files' directive.
+#
+# Core Functionality
+#   - Verify rpms package can be layered through rpm-ostree install
+#   - Verify multiple rpm packages can be layered through rpm-ostree install
+#   - Verify rpm packages can be removed through rpm-ostree uninstall
+#   - Verify multiple rpm packages can be removed through rpm-ostree uninstall
+#   - Verify rpm-ostree status output displays "Package: <pkg1> <pkg2>.." when packages are layered
+#   - Verify the -r / --reboot flag reboots the host after rpm-ostree install or rpm-ostree uninstall
+#   - Verify the -n / --dry-run flag does not perform the package add/remove function and displays a summary of packages to be installed
+#   - Verify a dry-run of rpm-ostree install|uninstall does not affect an existing package that has been layered, but not rebooted.
+#   - Verify that after additional repos are enabled (i.e. EPEL), rpm-ostree install|uninstall can be successfully used for packages in those repos.
+#
+# Negative Testing
+#   - Verify correct error is displayed removing a package that does not exist
+#   - Verify correct error is displayed when adding a package that does not exists in the repo
+#   - Verify correct error message is displayed when adding a package that is already in the deployment
+#   - Verify packages which have files with non-root ownership cannot be installed
+#   - Verify that install/uninstall operations cannot be performed by unprivileged users
+#
+
+- name: Package Layering Tests
+  hosts: all
+  become: yes
+
+  vars_files:
+    - vars.yml
+
+  pre_tasks:
+    - name: Verify all 
+      fail:
+        msg: "Required variables are not defined.  Check vars.yml file."
+      when: g_pkg1 is undefined or
+            g_pkg2 is undefined or
+            g_invalid_pkg is undefined or
+            g_nonroot_pkg is undefined or
+            g_deployed_pkg is undefined or
+            g_epel_pkg is undefined
+
+  roles:
+  # Subscribe if the system is RHEL
+    - role: redhat_subscription
+      when: ansible_distribution == 'RedHat'
+      tags:
+        - redhat_subscription
+
+    # single package install and uninstall
+    - { role: rpm_ostree_install, packages: "{{ g_pkg1 }}", reboot: false, tags: ['rpm_ostree_install11'] }
```

Comment:
Any reason for not unpacking these dicts on multiple lines to make it more legible?